src/dune_rules/dialect.ml

```diff
@@ -54,15 +54,16 @@ let to_dyn { name; file_kinds } =
   let open Dyn in
   record
     [ ("name", string name)
-    ; ("file_kinds", Ml_kind.Dict.to_dyn File_kind.to_dyn file_kinds)
+    ; ("file_kinds", Ml_kind.Dialect.to_dyn File_kind.to_dyn file_kinds)
     ]
 
 let encode { name; file_kinds } =
   let open Dune_lang.Encoder in
   let file_kind_stanzas =
     let open Ml_kind in
     List.map ~f:File_kind.encode
-      [ Dict.get file_kinds Intf; Dict.get file_kinds Impl ]
+    @@ List.filter_opt
```

Comment:
You should be able to refactor that as a `List.filter_map all ~f:(fun kind -> ...)`

Reply:
done